### PR TITLE
Introduce DTOs

### PR DESCRIPTION
**Pull request by ashleybroughton**
Anyone is welcome to submit PR's to my [fork](https://github.com/ashleybroughton/CoreWiki/tree/dto) for this pull request to keep work together.

We are looking to replace any direct references to entity objects with data transfer objects (DTO's).
If you are unsure what this means or how to do it, look at the `Files changes` tab above to get an idea.

Pages to convert to DTO

* [x] Create
* [x] CreateArticleFromLink
* [x] Delete
* [x] Details (GET)
* [x] Details (post - creating comments)
* [x] Edit
* [x] History
* [x] Search

### CODE REVIEW

File: CoreWiki/Pages/Details.cshtml.cs [hunk 1]
```diff
@@ -87,24 +114,23 @@ public class DetailsModel : PageModel
...
 
-			comment.Article = Article;
-
+			comment.Article = article;
 			comment.Submitted = _clock.GetCurrentInstant();
-
-			var author = await _UserManager.FindByIdAsync(Article.AuthorId.ToString());
 			await _commentRepo.CreateComment(comment);
 
+			// IDEA: Make this an extensibility module, we should only be creating a comment here (single responsibility principle)
```
Comment (erwindevreugd) [inline CoreWiki/Pages/Details.cshtml.cs:139]: Great idea! #224

**Comment by erwindevreugd** [comment]
What are your thoughts on what the data flow should be?

Repo->DataModel->Dto->PageModel/PageModel->Dto->DataModel->Repo?

or using the dto's as the page model?

**Comment by ashleybroughton** [comment]
Ready for review.  
Found a few bugs during testing which I have created issues for.  
Have confirmed via dev branch that they were not introduced by these changes.

**Comment by csharpfritz** [comment]
Thanks so much for this code... we really appreciate it

**Pull request merged**